### PR TITLE
Add ol.Object.unset() method

### PR DESCRIPTION
**Pull request by gberaudo**
This PR allows deletion of an object property.
Closes: #3404

### CODE REVIEW

File: src/ol/object.js [hunk 1]
```diff
@@ -163,6 +163,8 @@ ol.ObjectAccessor.prototype.transform = function(from, to) {
  * details, and see {@link ol.dom.Input} for the specific case of binding an
  * object with an HTML element.
  *
+ * Own observable properties may be removed by calling `unset()`.
```
Comment (elemoine) [inline src/ol/object.js]: I don't think we use the term "own observable property" elsewhere. And I don't really understand it. So I'd just say "Properties can be deleted by using the unset method. E.g. object.unset('foo')."

File: src/ol/object.js [hunk 2]
```diff
@@ -489,3 +491,17 @@ ol.Object.prototype.unbindAll = function() {
     this.unbind(key);
   }
 };
+
+
+/**
+ * Unset an observable own property.
```
Comment (elemoine) [inline src/ol/object.js]: -> Unset the property "key".

**Comment by elemoine** [comment]
I just added a comment regarding the vocabulary used in the docs. Otherwise looks good to me.

File: src/ol/object.js (hunk 1 above)
Comment (gberaudo) [inline src/ol/object.js]: The description of the class writes about `Classes that inherit from this
have pre-defined properties, to which you can add your own.`

You are right `own observable properties` is a bit too much.
What about using `own properties` to differenciate from the `pre-defined` properties?

File: src/ol/object.js [hunk 3]
```diff
@@ -489,3 +491,17 @@ ol.Object.prototype.unbindAll = function() {
...
+ * @param {string} key
```
Comment (elemoine) [inline src/ol/object.js]: Add a description.

```
 * @param {string} key The property key.
```

**Comment by gberaudo** [comment]
The PR was updated based on your comments. Thank you for the review @elemoine.

**Comment by elemoine** [comment]
Thanks. Will merge when the Travis build is finished.

**Pull request merged**